README.md

```diff
@@ -1,5 +1,88 @@
 Discovery
 =========
 
-TODO
+Weave Discovery retrieves and watches a list of peers
+available in a Weave network from a persistent storage.
+Discovery uses this information for telling a Weave Router
+about new peers it must connect to as well as peers
+that are not available anymore and must be forgotten.
+Discovery will also advertise the Weave router in the
+same backend so other peers can join it.
+
+The storage backends used by Discovery are compatible
+with [Docker Swarm](https://github.com/docker/swarm), so
+you could use Discovery for keeping a Swarm connected
+with a Weave network.
```

Comment:
is "persistent storage" and "storage backend" the right terminology here? "Storage" seems kinda weird. And I don't see swarm calling it that.